bug: main does not compile — pkg/extproc calls removed function streamingCachedPromptTokens

## Summary

`main` currently does not compile — `pkg/extproc` fails typecheck, which breaks `build-router` and cascades to failing CI on **every** open PR (`build_pr/*`, `test-and-build`, `component-benchmarks`, `integration-test`).

## Error

```
pkg/extproc/processor_res_body_streaming_response_api.go:320:19: undefined: streamingCachedPromptTokens
```

## Repro

```bash
cd src/semantic-router && go build ./pkg/extproc/
```

## Root cause

A semantic conflict between two recently-merged PRs. They touch different files, so git merged them with no text conflict, but the resulting tree calls a function that no longer exists — and the combined state was never CI-tested (#2438 merged on stale checks without rebasing onto post-#2430 `main`):

- **#2430** (GPT-5.6 cache-aware pricing, merged 2026-07-11) renamed/expanded
  `streamingCachedPromptTokens(ctx, promptTokens) (int, bool)`
  into
  `streamingPromptTokenDetails(ctx, promptTokens) (cached int, cachedReported bool, cacheWrite int, cacheWriteReported bool)`.
- **#2438** (translate responses api streaming chunks, merged 2026-07-12, ~19h later) was developed against an older base that still had the old function, and added a **new call** to `streamingCachedPromptTokens` in the new file `processor_res_body_streaming_response_api.go`.

## Fix

Update the call in `responseAPIStreamUsage` to use the new `streamingPromptTokenDetails`, taking the `cached` / `cachedReported` return values (behavior-preserving). PR incoming.

## Impact

Blocks all PRs until fixed. This class of break could be avoided by requiring branches to be up to date with `main` before merge (or using a merge queue that re-tests the combined result).

--- a/src/semantic-router/pkg/extproc/processor_res_body_streaming_response_api.go
+++ b/src/semantic-router/pkg/extproc/processor_res_body_streaming_response_api.go
@@ -317,7 +317,7 @@ func responseAPIStreamUsage(ctx *RequestContext) map[string]interface{} {
 
 	inputTokens := int(usage.PromptTokens)
 	cachedTokens := 0
-	if cached, ok := streamingCachedPromptTokens(ctx, inputTokens); ok {
+	if cached, cachedReported, _, _ := streamingPromptTokenDetails(ctx, inputTokens); cachedReported {
 		cachedTokens = cached
 	}
 

